src/routes/channel.js

```diff
@@ -51,29 +51,61 @@ router.post(
 
 /**
  * @route GET api/channels
- * @access Public
+ * @access Private
  * @desc Get all channels
  */
 router.get('/', async (req, res) => {
   try {
-    let allChannels = await Channel.find();
+    await Channel.find()
+      .populate('users')
+      .exec((err, channels) => {
+        channels.map(channel => {
+          return {
+            name: channel.name,
+            description: channel.description,
+            dateCreated: channel.date,
+            id: channel._id,
+            users: channel.users,
+            comments: channel.comments,
+          };
+        });
 
-    if (allChannels) {
-      const channels = allChannels.map(channel => {
-        return {
-          name: channel.name,
-          description: channel.description,
-          dateCreated: channel.date,
-          id: channel._id,
-        };
+        return res.status(200).send(channels);
       });
-
-      return res.status(200).send(channels);
-    }
   } catch (error) {
     console.log(error);
     res.status(500).json({ message: 'Something went wrong' });
   }
 });
 
+/**
+ * @route POST api/channels
+ * @access Private
+ * @desc Add the new user to the General channel
+ */
+router.post(
+  '/general',
+  [
+    check('users', 'User id is required')
+      .not()
+      .isEmpty(),
+  ],
+  (req, res) => {
+    Channel.find({ name: 'General' })
+      .then(generalChannel => {
+        Channel.findById(generalChannel[0]._id).then(channel => {
+          channel.users.push(req.body.users);
+
+          channel.save().then(channel => {
+            return res.status(200).send(`channel: ${channel}`);
+          });
+        });
+      })
+      .catch(error => {
+        console.log(error);
+        res.status(500).json({ message: 'Something went wrong' });
+      });
+  },
+);
```

Comment:
I think all this logic should be inside of the user router.  After we create the user we immediately find the `General` channel and add the user there.  Would this require you to make another API request after user registers just to get them into a channel?

Reply:
Good point. I'll move it to the user route.